src/extension.ts

```diff
@@ -112,30 +113,45 @@ function tryToForceLanguageId(document: TextDocument, fileName: string, document
         const oldLanguageId: string = document.languageId;
         languages.setTextDocumentLanguage(document, languageId);
         if (notifyUser && !documentCache.includes(document.fileName)) {
-          const CONFIGURE_IN_SETTINGS = "Configure in Settings";
-          const DISABLE_IN_SETTINGS = "Disable Language Updating";
-          const response: Thenable<string> = window.showInformationMessage(
-            `Quarkus Tools for Visual Studio Code automatically switched the language ID of '${fileName}' `
-            + `to be '${languageId}' in order to provide language support. `
-            + `This behavior can be configured in settings.`, DISABLE_IN_SETTINGS, CONFIGURE_IN_SETTINGS);
-          response.then(result => {
-            if (result === CONFIGURE_IN_SETTINGS) {
-              commands.executeCommand('workbench.action.openSettings', QuarkusConfig.PROPERTIES_LANGUAGE_MISMATCH);
-            } else if (result === DISABLE_IN_SETTINGS) {
-              QuarkusConfig.setPropertiesLanguageMismatch(PropertiesLanguageMismatch.ignore).then(() => {
-                languages.setTextDocumentLanguage(document, oldLanguageId);
-              });
-            }
-          });
-          documentCache.push(document.fileName);
+          const context = QuarkusContext.getExtensionContext();
+          if (!hasShownSetLanguagePopUp(context)) {
+            const CONFIGURE_IN_SETTINGS = "Configure in Settings";
+            const DISABLE_IN_SETTINGS = "Disable Language Updating";
+            const response: Thenable<string> = window.showInformationMessage(
+              `Quarkus Tools for Visual Studio Code automatically switched the language ID of '${fileName}' `
+              + `to be '${languageId}' in order to provide language support. `
+              + `This behavior can be configured in settings.`, DISABLE_IN_SETTINGS, CONFIGURE_IN_SETTINGS);
+            response.then(result => {
+              if (result === CONFIGURE_IN_SETTINGS) {
+                commands.executeCommand('workbench.action.openSettings', QuarkusConfig.PROPERTIES_LANGUAGE_MISMATCH);
+              } else if (result === DISABLE_IN_SETTINGS) {
+                QuarkusConfig.setPropertiesLanguageMismatch(PropertiesLanguageMismatch.ignore).then(() => {
+                  languages.setTextDocumentLanguage(document, oldLanguageId);
+                });
+                context.globalState.update(QuarkusConfig.QUARKUS_SET_LANGUAGE_ID, 'true');
```

Comment:
So if the user chooses `CONFIGURE_IN_SETTINGS`, the popup still gets shown every time ? Based on the name I figured it should stop showing in both cases.

Reply:
Should we make it more apparent with a "Don't Show Again" option?

Reply:
No, I think it should be automatic. Whichever option you click, the popup shouldn't be shown again.